added weapon stats for spectators

mexikoedi opened the pull request:
I adjusted the code to make the weapon stats visible for the spectators.
It still works for alive players but now also for spectators (tested it).
This fixes #2 .

Some things I noticed:
- local mat_tid_ammo = Material("vgui/ttt/tid/tid_ammo") is not used and there is no tid_ammo material in the materials folder, can I remove it (line 3)?
- There are no icons for weapons and grenades. (Normally there would be the icon for the "use" key. Is there a suitable icon for weapons and grenades? I would like to add icons there too.)
- There is some space between the title and the stats. (Normally there would be additional information which is useful for alive players like (walk over subtitle, hold "use" key, ...). Should I try to remove the space or add something there?)

Right now it looks like this:
- For weapons:
![targetid_weapon](https://github.com/user-attachments/assets/4b860fd4-3c24-48a9-9cba-04902717d4ca)

- For ammo boxes:
![targetid_ammo](https://github.com/user-attachments/assets/2973b3a9-4bc3-4bab-a163-51484fc606f9)

- For grenades:
![targetid_grenade](https://github.com/user-attachments/assets/9fbcff19-e81c-466a-9ada-c8f0e4c77663)

mexikoedi committed:
added weapon stats for spectators

TimGoll commented:
Not sure if I like that there is an icon for ammo, but not for the others I understand why you did it. The easiest solution would be to just hide the icon for specs as well

mexikoedi commented:
> Not sure if I like that there is an icon for ammo, but not for the others I understand why you did it. The easiest solution would be to just hide the icon for specs as well

Ok, I removed the ammo icon.
I also removed the unused code line.

mexikoedi committed:
removed ammo icon for spectators

TimGoll reviewed (approved):
thanks for improving the code
